### PR TITLE
Add types to ParameterService

### PR DESCRIPTION
**Pull request by InvincibleRMC**

### CODE REVIEW

File: rclpy/rclpy/parameter_service.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 import weakref
+from typing import TYPE_CHECKING
```
Comment (clalancette) [inline rclpy/rclpy/parameter_service.py:16]:
```suggestion
from typing import TYPE_CHECKING
import weakref
```

File: rclpy/rclpy/parameter_service.py [hunk 2]
```diff
@@ -65,7 +69,9 @@ def __init__(self, node):
             qos_profile=qos_profile_parameters
         )
 
-    def _describe_parameters_callback(self, request, response):
+    def _describe_parameters_callback(self, request: DescribeParameters.Request,
+                                      response: DescribeParameters.Response
+                                      ) -> DescribeParameters.Response:
```
Comment (clalancette) [inline rclpy/rclpy/parameter_service.py:74]: I don't particularly care for this style (even if it does pass our linters).  I think elsewhere we do:

```suggestion
    def _describe_parameters_callback(
        self,
        request: DescribeParameters.Request,
        response: DescribeParameters.Response
    ) -> DescribeParameters.Response:
```

(same goes for similar changes below)

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rclpy/rclpy/parameter_service.py (hunk 2 above)
Comment (InvincibleRMC) [reply rclpy/rclpy/parameter_service.py]: The linter seems to prefer

```python
def _describe_parameters_callback(
            self,
            request: DescribeParameters.Request,
            response: DescribeParameters.Response
            ) -> DescribeParameters.Response:
 ```
      
 over
      
 ```python
      def _describe_parameters_callback(
            self,
            request: DescribeParameters.Request,
            response: DescribeParameters.Response
          ) -> DescribeParameters.Response:
```

Comment (clalancette) [reply rclpy/rclpy/parameter_service.py]: See, for instance, https://github.com/ros2/rclpy/blob/adfcb2bd00a8301b310e4fd90731d61e94a86753/rclpy/rclpy/executors.py#L314-L318

Comment (InvincibleRMC) [reply rclpy/rclpy/parameter_service.py]: Ah I see.

**Review by clalancette — APPROVED** [review]
Looks good to me with green CI!

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21595/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15883/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1096/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22313/)

**Review by fujitatomoya — APPROVED** [review]
lgtm, i think a few test failures are unrelated.

**Comment by clalancette** [comment]
I've run the RHEL jobs a few times, and it always fails in the same way.  That makes me suspicious that this PR is causing it, as our nightlies don't seem to have that failure.  I'll try to look into it a bit more today.

**Review by sloretz — APPROVED** [review]

**Comment by sloretz** [comment]
Merged rolling into this branch to see if that resolves the issue on RHEL (the test failure on RHEL seemed unlikely to be caused by this PR to me)

Gist: https://gist.githubusercontent.com/sloretz/bf7bf128ff91ed2538b4608dc4d0554a/raw/85e809efe21359caedf415ff6cb14693cb2d34b2/ros2.repos
BUILD args:  --packages-up-to rclpy
TEST args:  --packages-select rclpy
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14358
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/21611/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/15899/)
* Linux-rhel [Build Status](http://ci.ros2.org/job/ci_linux-rhel/1116/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/22329/)

**Pull request merged**